tools/cldr/cldr-to-icu/src/test/java/org/unicode/icu/tool/cldrtoicu/mapper/Bcp47MapperTest.java

```diff
@@ -0,0 +1,281 @@
+
+        ImmutableList<IcuData> icuData = Bcp47Mapper.process(cldrData);
+        IcuData bcp47Data = icuData.get(0);
+
+        assertThat(bcp47Data).hasValuesFor("/keyMap/key-alias", "key");
+        assertThat(bcp47Data).hasValuesFor("/typeMap/key-alias/main", "type");
+        assertThat(bcp47Data).hasValuesFor("/typeAlias/key-alias/type-alias", "main");
+    }
+
+    @Test
+    public void testPreferredTypeName() {
+        CldrData cldrData = cldrData(
+            deprecated("deprecated-key", true, "type", false, "/preferred/path1"),
+            deprecated("key", false, "deprecated-type", true, "/preferred/path2"));
+
+        ImmutableList<IcuData> icuData = Bcp47Mapper.process(cldrData);
+        IcuData bcp47Data = icuData.get(0);
+
+        assertThat(bcp47Data).hasValuesFor("/bcpTypeAlias/deprecated-key/type", "/preferred/path1");
+        assertThat(bcp47Data).hasValuesFor("/bcpTypeAlias/key/deprecated-type", "/preferred/path2");
+    }
+
+    @Test
+    public void testInfoAttributes() {
+        CldrData cldrData = cldrData(
+            // Deprecated without a replacement.
+            deprecated("deprecated-key", true, "type", false, null),
+            deprecated("key", false, "deprecated-type", true, null),
+            valueType("info-key", "info-type", "value-type"));
+
+        ImmutableList<IcuData> icuData = Bcp47Mapper.process(cldrData);
+        IcuData bcp47Data = icuData.get(0);
+
+        assertThat(bcp47Data).hasEmptyValue("/keyMap/deprecated-key");
+        assertThat(bcp47Data).hasEmptyValue("/typeMap/deprecated-key/type");
+        assertThat(bcp47Data).hasValuesFor("/keyInfo/deprecated/deprecated-key", "true");
+
+        assertThat(bcp47Data).hasEmptyValue("/keyMap/key");
+        assertThat(bcp47Data).hasEmptyValue("/typeMap/key/deprecated-type");
+        assertThat(bcp47Data).hasValuesFor("/typeInfo/deprecated/key/deprecated-type", "true");
+
+        assertThat(bcp47Data).hasEmptyValue("/keyMap/info-key");
+        assertThat(bcp47Data).hasEmptyValue("/typeMap/info-key/info-type");
+        assertThat(bcp47Data).hasValuesFor("/keyInfo/valueType/info-key", "value-type");
+    }
+
+    // This will hopefully one day be the responsibility of the IcuTextWriter.
+    @Test
+    public void testTimezonePathQuotingForAliases() {
+        CldrData cldrData = cldrData(
+            alias("tz", null, "escaped", "foo/bar", "hello/world"),
+            alias("tz", null, "unescaped", "foo_bar", "hello_world"));
+
+        ImmutableList<IcuData> icuData = Bcp47Mapper.process(cldrData);
+        IcuData tzData = icuData.get(1);
+
+        // Only the type-map paths/values are split into the timezone data.
+        assertThat(tzData).getPaths().hasSize(4);
+        assertThat(tzData).hasValuesFor("/typeMap/tz/\"foo:bar\"", "escaped");
+        assertThat(tzData).hasValuesFor("/typeMap/tz/foo_bar", "unescaped");
+
+        // TODO: Check this is correct! Having foo/bar as the value rather than foo:bar seems wrong.
```

Comment:
@macchiati @markusicu - Please have a look at this since I think it's doing exactly what the old converter did (but perhaps this case was never hit or should not be hit). Either way it feels wrong that the path now contains "foo:bar" above, but the alias "name" is "foo/bar". Am happy to fix, or comment one way or the other.